ui/components/multichain/multi-srp/srp-list/srp-list.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  KeyringMetadata,
+  KeyringObject,
+  KeyringTypes,
+} from '@metamask/keyring-controller';
+import { InternalAccount } from '@metamask/keyring-internal-api';
+import Card from '../../../ui/card';
+import {
+  Box,
+  IconName,
+  Icon,
+  Text,
+  IconSize,
+  AvatarAccount,
+  AvatarAccountSize,
+} from '../../../component-library';
+import {
+  JustifyContent,
+  Display,
+  TextColor,
+  FlexDirection,
+  AlignItems,
+  BlockSize,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import {
+  getMetaMaskCachedBalances,
+  getMetaMaskKeyrings,
+} from '../../../../selectors/selectors';
+import { getInternalAccounts } from '../../../../selectors/accounts';
+import UserPreferencedCurrencyDisplay from '../../../app/user-preferenced-currency-display/user-preferenced-currency-display.component';
+import { shortenAddress } from '../../../../helpers/utils/util';
+import { Numeric } from '../../../../../shared/modules/Numeric';
+import { EtherDenomination } from '../../../../../shared/constants/common';
+import { useMultichainSelector } from '../../../../hooks/useMultichainSelector';
+import { getMultichainConversionRate } from '../../../../selectors/multichain';
+
+type KeyringObjectWithMetadata = KeyringObject & { metadata: KeyringMetadata };
+
+export const SRPList = ({
+  onActionComplete,
+  hideShowAccounts,
+}: {
+  onActionComplete: (id: string) => void;
+  hideShowAccounts?: boolean;
+}) => {
+  const keyrings: KeyringObjectWithMetadata[] =
+    useSelector(getMetaMaskKeyrings);
+  const accounts: InternalAccount[] = useSelector(getInternalAccounts);
+  const accountBalances: Record<string, string> = useSelector(
+    getMetaMaskCachedBalances,
+  );
```

Comment:
This only uses EVM accounts IIRC. This won't work for Solana, and AFAIK we don't have a selector that combines both EVM and non-EVM balances.